dts/bindings/spi/microchip,mec5-qspi.yaml

```diff
@@ -0,0 +1,79 @@
+# Copyright (c) 2024, Microchip Technology Inc.
+# SPDX-License-Identifier: Apache-2.0
+
+description: Microchip MEC5 series QSPI controller
+
+compatible: "microchip,mec5-qspi"
+
+include: [spi-controller.yaml, pinctrl-device.yaml]
+
+properties:
+  reg:
+    required: true
+
+  pinctrl-0:
+    required: true
+
+  pinctrl-names:
+    required: true
+
+  lines:
+    type: int
+    description: |
+      QSPI data lines 1, 2, or 4. 1 data line is full-duplex
+      MOSI and MISO or half-duplex on MOSI only. Lines set to 2
+      or 4 indicate dual or quad I/O modes.
+      Defaults to 1 for full duplex driver's support for full-duplex spi.
+    enum:
+      - 1
+      - 2
+      - 4
+
+  chip-select:
+    type: int
+    description: |
+      Use QSPI CS0# or CS1#. Shared SPI supports two chip selects.
+      All other QSPI ports support one chip select.
+
```

Comment:
all the attributes below are specific for the peripheral connected to the controller, so it dos not belong here.

Reply:
@tbursztyka Hi, I need some clarification.
1. Are you saying all the YAM properties after chip-select belong to a microchip specific flash device YAML?
2. Chip-select can come from the reg property of child flash devices but the SPI DT macros are not using the reg property.  SPI DT macros are using the SPI_CS_CONTROL_INIT and then GPIO DT spec. This SPI controller can't use arbitrary GPIO pins, its HW design requires it control the CS pin.  So I still need to application to create cs-gpios DT object and my driver does not call GPIO driver but derives the cs number (0 or 1) from cs-gpios?

Reply:
SPI DT macros save the SPI child device "reg" property value in the SPI configuration struct slave member. This driver will use the child device "reg" property as the chip select. MCHP QSPI is controller mode only hardware.